package-res/resources/web/test/karma/unit/pentaho/type/string.Spec.js

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+    "pentaho/type/string",
+    "pentaho/type/Context",
+    "pentaho/i18n!/pentaho/type/i18n/types"
+], function(stringFactory, Context, bundle) {
+
+    "use strict";
+
+    /*global describe:true, it:true, expect:true, beforeEach:true*/
+
+    describe("pentaho/type/string -", function() {
+        it("is a function", function() {
+            expect(typeof stringFactory).toBe("function");
+        });
+
+        describe("new string()", function() {
+            var PentahoString;
+            beforeEach(function () {
+                PentahoString = stringFactory(new Context());
+            });
+            it("should return an object", function () {
+                expect(typeof new PentahoString("string")).toBe("object");
+            });
+            it("should return a function", function () {
+                expect(typeof PentahoString).toBe("function");
+            });
+            it("should accept 1 as '1'", function () {
+                expect(new PentahoString(1)._value).toBe('1');
+            });
+            it("should accept '1' as '1'", function () {
+                expect(new PentahoString('1')._value).toBe('1');
+            });
+            it("should accept true as 'true'", function () {
+                expect(new PentahoString(true)._value).toBe('true');
+            });
+            it("should accept 'true' as 'true'", function () {
+                expect(new PentahoString('true')._value).toBe('true');
+            });
+            it("should accept empty string as ''", function () {
+                expect(new PentahoString('')._value).toBe('');
+            });
+            it("should not accept null", function () {
+                expect(function () {
+                    new PentahoString(null)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+            it("should not accept undefined", function () {
+                expect(function () {
+                    new PentahoString(undefined)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+        });
```

Comment:
`PentahoString({}).value` should throw cause it is taken as a spec and the value is not there, thus it is `null` :-)